Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -103,30 +94,43 @@ public async Task<FileExistenceResult> CheckFileExistsAsync(Context context, Con
         /// <summary>
         /// Copies content from the server to the given local path.
         /// </summary>
-        public Task<CopyFileResult> CopyFileAsync(Context context, ContentHash hash, AbsolutePath destinationPath, CancellationToken ct = default(CancellationToken))
+        public Task<CopyFileResult> CopyFileAsync(Context context, ContentHash hash, AbsolutePath destinationPath, CancellationToken ct)
         {
             Func<Stream> streamFactory = () => new FileStream(destinationPath.Path, FileMode.Create, FileAccess.Write, FileShare.None, _bufferSize, FileOptions.SequentialScan);
-            return CopyToAsync(context, hash, streamFactory, ct);        
+
+            return CopyToAsync(context, hash, streamFactory, ct);
         }
         
         /// <summary>
         /// Copies content from the server to the given stream.
         /// </summary>
-        public Task<CopyFileResult> CopyToAsync(Context context, ContentHash hash, Stream stream, CancellationToken ct = default(CancellationToken))
+        public Task<CopyFileResult> CopyToAsync(Context context, ContentHash hash, Stream stream, CancellationToken ct)
         {
             return CopyToAsync(context, hash, () => stream, ct);
         }
 
         /// <summary>
         /// Copies content from the server to the stream returned by the factory.
         /// </summary>
-        public async Task<CopyFileResult> CopyToAsync(Context context, ContentHash hash, Func<Stream> streamFactory, CancellationToken ct = default(CancellationToken))
+        public async Task<CopyFileResult> CopyToAsync(Context context, ContentHash hash, Func<Stream> streamFactory, CancellationToken ct)
+        {
+            // Need to track shutdown to prevent invalid operation errors when the instance is used after it was shut down is called.
+            using (var operationContext = TrackShutdown(context, ct))
+            {
+                return await CopyToCoreAsync(operationContext, hash, streamFactory);

```

Comment:
>await [](start = 23, length = 5)

isn't this await superfluous if you also remove the async?

Reply:
It is very much needed here because it is happening inside the using block.

Without it `operationContext` will be disposed immediately without waiting for the copy to finish.

Reply:
I make this mistake again and again. So nice to see Danny make it too.